Use noarch Python

jakirkham opened the pull request:
Refresh the recipe a bit and convert it to `noarch: python`. More details in the commit messages.

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham commented:
@conda-forge-admin, please re-render.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I rerendered the feedstock and it seems to be already up-to-date.

jakirkham committed:
Switch to `sha256`

As a hash, `md5` is pretty broken. So switch to `sha256`, which is not
yet broken.

jakirkham committed:
Package the license file

Make sure to package the license file from the source for end user
inspection.

jakirkham committed:
Change `build` to `host`

The build environment where everything gets installed into is now called
`host`. So update the name in `requirements` accordingly.

jakirkham committed:
Flatten `script` install line

jakirkham committed:
Use `pip` for build and install

Make sure to use `pip` to do the build and install as that is the
direction the Python ecosystem is going. Also makes it easier for
`conda` packages to play nicely with `pip` packages.

jakirkham committed:
Convert to `noarch: python`

As the package is pure Python with no conditional constraints, switch
the package to `noarch: python` to make it easier to maintain and make
installation work on a large range of Pythons and OSes.

jakirkham committed:
MNT: Re-rendered with conda-smithy 3.1.6 and pinning 2018.07.01

Now that this is `noarch: python`, re-render to update the variants,
CIs, and README to reflect this change.

jakirkham committed:
Bump build number to 1

Rebuild the package now that it uses `pip` and `noarch: python`.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.
